Mods/SML/Source/SML/Private/Registry/ModContentRegistry.cpp

```diff
@@ -860,13 +860,39 @@ void UModContentRegistry::AutoRegisterRecipeReferences( TSubclassOf<UFGRecipe> R
 	}
 }
 
+static bool IsResourceFormFilteredOut(EResourceForm ResourceForm, EGetObtainableItemDescriptorsFlags Flags)
+{
+	using enum EGetObtainableItemDescriptorsFlags;
+	using enum EResourceForm;
+
+	switch (ResourceForm) {
+	case RF_INVALID:
+		return !EnumHasAnyFlags(Flags, EGetObtainableItemDescriptorsFlags(IncludeBuildings | IncludeCustomizations | IncludeVehicles | IncludeCreatures | IncludeSpecial));
+	case RF_SOLID:
+		return false;
+	case RF_LIQUID:
+	case RF_GAS:
+		return !EnumHasAnyFlags(Flags, EGetObtainableItemDescriptorsFlags(IncludeNonSolid));
+	case RF_HEAT: /* FIXME: Could be omitted, so that the default case handles it */
```

Comment:
Should we omit this? It would be handled by the `default` case, which filters out unknown resource forms.